spec.html

```diff
@@ -47528,10 +47528,10 @@ <h1>String.prototype.substr ( _start_, _length_ )</h1>
           1. Let _intStart_ be ? ToIntegerOrInfinity(_start_).
           1. If _intStart_ is -&infin;, set _intStart_ to 0.
           1. Else if _intStart_ &lt; 0, set _intStart_ to max(_size_ + _intStart_, 0).
+          1. Else, set _intStart_ to min(_intStart_, _size_).
```

Comment:
Can't we use clamping here, too?

```suggestion
          1. If _intStart_ &lt; 0, set _intStart_ to _size_ + _intStart_.
          1. Set _intStart_ to the result of clamping _intStart_ between 0 and _size_.
```

Reply:
That does arithmetic on an infinite value (-∞), which we've tried to avoid. If you restore the first line then the other changes are fine, though I don't think it's any clearer or more elegant, personally.

Reply:
And in addition to the `-∞` issue, the current PR matches how [String.prototype.slice](https://tc39.es/ecma262/#sec-string.prototype.slice) is specified. If we should decide to use clamping, we should also update `String.prototype.slice` accordingly.

Reply:
I still think the clamping is more clear, but @bakkot is right, we should leave the guard in for infinity. We can update `String.prototype.slice` in a separate PR if you want.

Reply:
@michaelficarra If you approve this as-is I will submit a following changing both; no need to drag this PR out further.